packages/grpc-native-core/index.d.ts

```diff
@@ -0,0 +1,1214 @@
+     * write, or until writes_done, but it need not buffer completely or at all.
+     */
+    BUFFER_HINT = 1,
+    /**
+     * Force compression to be disabled for a particular write
+     */
+    NO_COMPRESS,
+  }
+
+  /**
+   * Log verbosity constants. Maps setting names to code numbers.
+   */
+  export enum logVerbosity {
+    DEBUG,
+    INFO,
+    ERROR,
+  }
+
+  /**
+   * Credentials module
+   *
+   * This module contains factory methods for two different credential types:
+   * CallCredentials and ChannelCredentials. ChannelCredentials are things like
+   * SSL credentials that can be used to secure a connection, and are used to
+   * construct a Client object. CallCredentials genrally modify metadata, so they
+   * can be attached to an individual method call.
+   *
+   * CallCredentials can be composed with other CallCredentials to create
+   * CallCredentials. ChannelCredentials can be composed with CallCredentials
+   * to create ChannelCredentials. No combined credential can have more than
+   * one ChannelCredentials.
+   *
+   * For example, to create a client secured with SSL that uses Google
+   * default application credentials to authenticate:
+   *
+   * ```
+   * var channel_creds = credentials.createSsl(root_certs);
+   * (new GoogleAuth()).getApplicationDefault(function(err, credential) {
+   *   var call_creds = credentials.createFromGoogleCredential(credential);
+   *   var combined_creds = credentials.combineChannelCredentials(
+   *       channel_creds, call_creds);
+   *   var client = new Client(address, combined_creds);
+   * });
+   * ```
+   */
+  export const credentials: {
+    /**
+     * Create an SSL Credentials object. If using a client-side certificate, both
+     * the second and third arguments must be passed.
+     * @param rootCerts The root certificate data
+     * @param privateKey The client certificate private key, if applicable
+     * @param certChain The client certificate cert chain, if applicable
+     * @return The SSL Credentials object
+     */
+    createSsl(rootCerts?: Buffer, privateKey?: Buffer, certChain?: Buffer): ChannelCredentials;
+
+    /**
+     * Create a gRPC credentials object from a metadata generation function. This
+     * function gets the service URL and a callback as parameters. The error
+     * passed to the callback can optionally have a 'code' value attached to it,
+     * which corresponds to a status code that this library uses.
+     * @param metadataGenerator The function that generates metadata
+     * @return The credentials object
+     */
+    createFromMetadataGenerator(metadataGenerator: metadataGenerator): CallCredentials;
+
+    /**
+     * Create a gRPC credential from a Google credential object.
+     * @param googleCredential The Google credential object to use
+     * @return The resulting credentials object
+     */
+    createFromGoogleCredential(googleCredential: GoogleCredential): CallCredentials;
```

Comment:
I don't believe that `GoogleCredential` is defined in this file, which is not surprising because gRPC does not explicitly depend on the library that that type comes from. That library is `google-auth-library`. I think it would be OK to depend just on the type definitions for that library, if such a module exists.

Reply:
I made a copy of the types into this file, see [line 864](https://github.com/grpc/grpc-node/pull/52/files/bdbbd5324956a118dacf21380a5e3d9eac6be62e#diff-84a425dc635cd3214797bf694fad81e8R864).
The lib is written in TypeScript so it ships with it's own typings (https://github.com/google/google-auth-library-nodejs/blob/master/package.json#L10)
How would you prefer this? I dep to the lib or a copy of the type?

Reply:
I see that copy now, and I think that's the right way to handle this. But that's not the right type. I think what I called `GoogleCredentials` actually corresponds to what's listed as `OAuth2Client` in that library. The important part of that interface, from the point of view of how this library uses it, is the `getRequestMetadata` method.

Reply:
Oh, that's my mistake then. I'll change it to the right one.

I've put som questions into the other comments you left, but I think they've been hidden by the other changes. Would you look into them as well?